/*
 * G2S web API
 * A Genome to Strucure (G2S) API Supports Automated Mapping and Annotating Genomic Variants in 3D Protein Structures. Supports Inputs from Human Genome Position, Uniprot and Human Ensembl Names.
 *
 * OpenAPI spec version: 1.0 (beta)
 * Contact: wangjue@missouri.edu
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package org.genomenexus.g2s.client.api;

import org.genomenexus.g2s.client.ApiException;
import org.genomenexus.g2s.client.model.Alignment;
import org.junit.Test;
import org.junit.Ignore;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * API tests for GetResidueMappingApi
 */
@Ignore
public class GetResidueMappingApiTest {

    private final GetResidueMappingApi api = new GetResidueMappingApi();

    
    /**
     * Get PDB Residue Mapping by Protein Sequence and Residue position
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void getPdbAlignmentReisudeBySequenceUsingGETTest() throws ApiException {
        String sequence = null;
        List<String> positionList = null;
        List<String> paramList = null;
        List<Alignment> response = api.getPdbAlignmentReisudeBySequenceUsingGET(sequence, positionList, paramList);

        // TODO: test validations
    }
    
    /**
     * Get PDB Residue Mapping by Protein Sequence and Residue position
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void getPdbAlignmentReisudeBySequenceUsingPOSTTest() throws ApiException {
        String sequence = null;
        List<String> positionList = null;
        List<String> paramList = null;
        List<Alignment> response = api.getPdbAlignmentReisudeBySequenceUsingPOST(sequence, positionList, paramList);

        // TODO: test validations
    }
    
    /**
     * Post Residue Mapping by ProteinId, PDBId and Chain
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void postResidueMappingByPDBUsingGETTest() throws ApiException {
        String idType = null;
        String id = null;
        String pdbId = null;
        String chainId = null;
        List<String> positionList = null;
        List<Alignment> response = api.postResidueMappingByPDBUsingGET(idType, id, pdbId, chainId, positionList);

        // TODO: test validations
    }
    
    /**
     * Post Residue Mapping by ProteinId, PDBId and Chain
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void postResidueMappingByPDBUsingPOSTTest() throws ApiException {
        String idType = null;
        String id = null;
        String pdbId = null;
        String chainId = null;
        List<String> positionList = null;
        List<Alignment> response = api.postResidueMappingByPDBUsingPOST(idType, id, pdbId, chainId, positionList);

        // TODO: test validations
    }
    
    /**
     * POST PDB Residue Mapping by ProteinId
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void postResidueMappingUsingGETTest() throws ApiException {
        String idType = null;
        String id = null;
        List<String> positionList = null;
        List<Alignment> response = api.postResidueMappingUsingGET(idType, id, positionList);

        // TODO: test validations
    }
    
    /**
     * POST PDB Residue Mapping by ProteinId
     *
     * 
     *
     * @throws ApiException
     *          if the Api call fails
     */
    @Test
    public void postResidueMappingUsingPOSTTest() throws ApiException {
        String idType = null;
        String id = null;
        List<String> positionList = null;
        List<Alignment> response = api.postResidueMappingUsingPOST(idType, id, positionList);

        // TODO: test validations
    }
    
}
